plugins/MauticCrmBundle/Integration/PipedriveIntegration.php

```diff
@@ -111,75 +111,81 @@ public function isAuthorized()
     }
 
     /**
-     * Get available company fields for choices in the config UI.
-     *
-     * @param array $settings
-     *
-     * @return array
-     */
-    public function getFormCompanyFields($settings = [])
-    {
-        $fields = $this->getAvailableLeadFields(self::ORGANIZATION_ENTITY_TYPE);
-
-        return $fields;
-    }
-
-    /**
-     * @param array $settings
-     *
      * @return array|mixed
      */
-    public function getFormLeadFields($settings = [])
+    public function getAvailableLeadFields($settings = [])
     {
-        $fields = $this->getAvailableLeadFields(self::PERSON_ENTITY_TYPE);
-
-        // handle fields with are available in Pipedrive, but not listed
-        return array_merge($fields, [
-            'last_name' => [
-                'label'    => 'Last Name',
-                'required' => true,
-            ],
-            'first_name' => [
-                'label'    => 'First Name',
-                'required' => true,
-            ],
-        ]);
-    }
-
-    /**
-     * @return array|mixed
-     */
-    public function getAvailableLeadFields($object = null)
-    {
-        $integrationFields = [];
-
-        /**
-         * $object as Array comes from clicking "Apply" button on Plugins Configuration form.
-         * I dont't know why its calling again pipedrive API to get fields which are already inside form...
-         * Also i have no idea why is trying to pass some strange object...
-         */
-        if (!$this->isAuthorized() || !$object || is_array($object)) {
-            return $integrationFields;
+        $pipedriveFields      = [];
+        $silenceExceptions    = (isset($settings['silence_exceptions'])) ? $settings['silence_exceptions'] : true;
+        if (isset($settings['feature_settings']['objects'])) {
+            $pipedriveObjects = $settings['feature_settings']['objects'];
+        } else {
+            $settings                                = $this->settings->getFeatureSettings();
+            $settings['feature_settings']['objects'] = $pipedriveObjects = isset($settings['objects']) ? $settings['objects'] : ['contacts'];
```

Comment:
Is the migration necessary considering this check?

Reply:
@alanhartless any feedback ? Thanks !

Reply:
@alanhartless  I would like remove contacts object as default until you check this object in Pipedrive plugin settings

https://github.com/mautic/mautic/pull/5746/files#diff-6ad058feeec6f77c7840ec788b1b26e1R124

Or do think contacts should be force enabled and then we can remove contacts object from settings  (there will no option for  choose contacts object, its mandatory  have contacts fields matching setup)